engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -600,6 +605,19 @@ public void activate()
       CHECKPOINT_WINDOW_COUNT = 1;
     }
 
+    int dagChkptWndwCnt = context.getValue(Context.DAGContext.CHECKPOINT_WINDOW_COUNT);
+    if (PROCESSING_MODE != ProcessingMode.EXACTLY_ONCE) {
+      int chkOffset = dagChkptWndwCnt % CHECKPOINT_WINDOW_COUNT;
+      if (chkOffset != 0) {
+        EFFECTIVE_CHECKPOINT_WINDOW_COUNT = dagChkptWndwCnt + CHECKPOINT_WINDOW_COUNT - chkOffset;
+      } else {
+        EFFECTIVE_CHECKPOINT_WINDOW_COUNT = dagChkptWndwCnt;
+      }
+    } else {
+      EFFECTIVE_CHECKPOINT_WINDOW_COUNT = 1;
+    }
```

Comment:
@PramodSSImmaneni Should EFFECTIVE_CHECKPOINT_WINDOW_COUNT and CHECKPOINT_WINDOW_COUNT not be same value? There are couple of places where ++checkpointWindowCount == CHECKPOINT_WINDOW_COUNT is checked to make checkpoint call